packages/documentation-v7/src/stories/components/switch/switch.stories.ts

```diff
@@ -0,0 +1,225 @@
+      },
+      control: {
+        type: 'text',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    labelAfter: {
+      name: 'Label (after)',
+      description: 'Describes the content/topic of the component.',
+      if: {
+        arg: 'labelPosition',
+        neq: 'before',
+      },
+      control: {
+        type: 'text',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    hiddenLabel: {
+      name: 'Hidden Label',
+      description:
+        'Renders the component with or without a visible label.<span className="mt-mini alert alert-info alert-sm">There are accessibility concerns with hidden labels.<br/>Please read our <a href="/?path=/story/foundations-accessibility--page#labels">label accessibility guide</a>.</span>',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    checked: {
+      name: 'Checked',
+      description: 'When set to `true`, places the component in the checked state.',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    disabled: {
+      name: 'Disabled',
+      description:
+        'When set to `true`, disables the component\'s functionality and places it in a disabled state.<span className="mt-mini alert alert-info alert-sm">There are accessibility concerns with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</span>',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    validation: {
+      name: 'Validation',
+      description: "Controls the display of the component's validation state.",
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid',
+        },
+      },
+      options: ['null', 'is-valid', 'is-invalid'],
+      table: {
+        category: 'States',
+      },
+    },
+  },
+  decorators: [
+    story =>
+      html`
+        <div class="pt-3">${story()}</div>
+      `,
+  ],
+};
+
+export default meta;
+
+const VALIDATION_STATE_MAP: Record<string, undefined | boolean> = {
+  'null': undefined,
+  'is-valid': false,
+  'is-invalid': true,
+};
+
+function renderSwitch(args: Args, context: StoryContext) {
+  const [_, updateArgs] = useArgs();
+
+  const switchClasses = mapClasses({
+    'form-check-input': true,
+    [args.validation]: args.validation !== 'null',
+  });
+
+  const useAriaLabel = args.hiddenLabel;
+  const useLabelBefore = !useAriaLabel && ['before', 'both'].includes(args.labelPosition);
+  const useLabelAfter = !useAriaLabel && ['after', 'both'].includes(args.labelPosition);
+
+  const labelBefore = useLabelBefore
+    ? html`
+        <label for="switch" class="form-check-label order-first">${args.labelBefore}</label>
```

Comment:
The label `for` attribute has to match the input's `id`, otherwise they are not connected. The user can't click on the label to toggle the switch and a screen reader does not know what label to read when the user is focusing on the input field.

```suggestion
        <label for=${context.id} class="form-check-label order-first">${args.labelBefore}</label>
```